src/nvim/terminal.c

```diff
@@ -1321,7 +1321,7 @@ static void adjust_topline(Terminal *term, buf_T *buf, long added)
       if (following || (wp == curwin && is_focused(term))) {
         // "Follow" the terminal output
         wp->w_cursor.lnum = ml_end;
-        set_topline(wp, MAX(wp->w_cursor.lnum - height + 1, 1));
+        set_topline(wp, MAX(wp->w_cursor.lnum - wp->w_height_inner + 1, 1));
```

Comment:
@bfredl 	does this make sense? This PR originally used `w_height` but that fails the  `test/functional/terminal/window_spec.lua` test.

Reply:
yes, makes sense.

Reply:
Thanks. Is there a qualitative/semantic explanation somewhere of the difference between `w_height` and `w_height_inner` (and `w_height_request`)? 

For a single-grid UI is `w_height` always the same as `w_height_inner`? Then it makes sense that `terminal/window_spec.lua` failed, because it has multiple grids. But I don't understand when `w_height` would ever be preferred to `w_height_inner`.

Reply:
> when w_height would ever be preferred to w_height_inner.

For any calculation involving the split layout. But It could be that it is redundant with `wp->w_frame->fr_height`, in case it could be eliminated.